src/platform/baytrail/lib/dma.c

```diff
@@ -119,32 +119,48 @@ const struct dw_drv_plat_data dmac2 = {
 };
 #endif
 
+/*
+ * Baytrail and Cherrytrail should use the folowing DMA mappings.
+ *
+ * Init DMAC0 for Mem2Phe and Phe2Mem Transfers
+ *
+ *    Channel 0 - SSP2 to Mem - Src_Id  = 0
+ *    Channel 1 - Mem to SSP2 - Dest_Id = 5
+ *    Channels 2-7 - Unused
+ *
+ * Init DMAC1 for Mem2Mem Transfers
+ *
+ *    Channel 0 - Unused
+ *    Channel 1 - Unused
+ *    Channels 2-7 - Mem to Mem
+ *
+ */
 SHARED_DATA struct dma dma[PLATFORM_NUM_DMACS] = {
 {
+	/* DMAC 0 Mem2Phe and Phe2Mem Transfers */
 	.plat_data = {
 		.id		= DMA_ID_DMAC0,
-		.dir		= DMA_DIR_MEM_TO_MEM | DMA_DIR_MEM_TO_DEV |
-				  DMA_DIR_DEV_TO_MEM | DMA_DIR_DEV_TO_DEV |
-				  DMA_DIR_HMEM_TO_LMEM | DMA_DIR_LMEM_TO_HMEM,
+		.dir		= DMA_DIR_MEM_TO_DEV |
+				  DMA_DIR_DEV_TO_MEM | DMA_DIR_DEV_TO_DEV,
 		.caps		= DMA_CAP_GP_HP,
-		.devs		= DMA_DEV_SSP | DMA_DEV_DMIC | DMA_DEV_HOST,
+		.devs		= DMA_DEV_SSP,
 		.base		= DMA0_BASE,
-		.channels	= 8,
+		.channels	= 2, /* Channels 2-7 - Unused */
 		.irq		= IRQ_NUM_EXT_DMAC0,
 		.drv_plat_data	= &dmac0,
 	},
 	.ops		= &dw_dma_ops,
 },
 {
+	/*  DMAC1 for Mem2Mem Transfers */
 	.plat_data = {
 		.id		= DMA_ID_DMAC1,
-		.dir		= DMA_DIR_MEM_TO_MEM | DMA_DIR_MEM_TO_DEV |
-				  DMA_DIR_DEV_TO_MEM | DMA_DIR_DEV_TO_DEV |
+		.dir		= DMA_DIR_MEM_TO_MEM |
 				  DMA_DIR_HMEM_TO_LMEM | DMA_DIR_LMEM_TO_HMEM,
 		.caps		= DMA_CAP_GP_HP,
-		.devs		= DMA_DEV_SSP | DMA_DEV_DMIC | DMA_DEV_HOST,
```

Comment:
any idea why we references DMICs here? It was never supported on this platform.
Wondering we should fix the .devs and .dir first, and in a different patch try the channels mapping in addition.

Reply:
Updated as I also had a spelling fix.